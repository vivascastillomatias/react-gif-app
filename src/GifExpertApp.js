import React, {useState} from 'react'
import { AddCategory } from './components/AddCategory'
import GifGrid from './components/GifGrid'

export default function GifExpertApp({ defaultCategories = ['homero']}) {

    const [categories, setCategories] = useState(defaultCategories)
    return (

        <>
            <h2>GifExpertApp ! - Curso de React Fernando Herrera</h2>
            <hr/>
            <AddCategory 
                setCategories = {setCategories}
            />
            <hr/>
            <ol>
                {
                    categories.map( (c) =>  (

                    <GifGrid 
                        key={c}
                        category={c}
                    /> 
                    ))
                }
            </ol>
        </>
    )
}
