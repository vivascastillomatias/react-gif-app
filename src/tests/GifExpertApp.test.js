import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Pruebas en <GifExpertApp />', () => {
    
    test('ConstrucciÃ³n correcta del componente', () => {
        
        const wrapper = shallow(<GifExpertApp />)

        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe mostrar una lista de categorias', () => {
        const categories = ['Homero', 'Bad Bunny']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
    
})
