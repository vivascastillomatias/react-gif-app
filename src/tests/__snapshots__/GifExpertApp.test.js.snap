// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Construcci√≥n correcta del componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp ! - Curso de React Fernando Herrera
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="homero"
      key="homero"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp ! - Curso de React Fernando Herrera
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Homero"
      key="Homero"
    />
    <GifGrid
      category="Bad Bunny"
      key="Bad Bunny"
    />
  </ol>
</Fragment>
`;
